### ISSUE TITLE
Migration to MIME based rendering

### ISSUE THREAD

**Issue by ellisonbg** [altair-viz/altair#283]
For a while, we have been wanting to move towards MIME type based rendering. In this approach, Altair would emit JSON data in the proper vega MIME type and the notebook, jupyterlab or nteract would know know to render the data appropriately. This is very different from the current approach used in ipyvega, where we actually send back JavaScript code for rendering. JupyterLab has supported this for a while, but the classic notebook just got this ability in the recent 4.3 release.

Here are the steps needed:

* Update the `jupyterlab_vega` package to support latest jupyterlab and notebook 4.3
* Update altair to require jupyterlab_vega and notebook 4.3.
* Change the display of the Chart objects in Altair to use this MIME based approach.

I will be working on this stuff in the coming week. A few questions:

* Do we just deprecate ipyvega completely?
* How do we version this stuff to make the transition easiest for users?

@jakevdp

**Comment by ellisonbg** [comment]
Related to #172 but broader as we want to also use this approach in the classic notebook.

**Comment by gnestor** [comment]
I created a fork of jupyterlab_vega and rewrote it using the [mimerender-cookiecutter](https://github.com/gnestor/mimerender-cookiecutter). It renders Vega and Vega-lite in both lab and notebook. I was testing and running into an issue when trying to render output from Altair and realized that Altair actually depends on the jupyter-vega extension to render Vega. The mimerender approach seems like a better solution. I created a PR for reference/merging: https://github.com/altair-viz/jupyterlab_vega/pull/3
